server/src/main/java/org/apache/cassandra/sidecar/livemigration/LiveMigrationFileDownloader.java

```diff
@@ -0,0 +1,688 @@
+            {
+                continue;
+            }
+
+            if (Files.isDirectory(localPath))
+            {
+                LOGGER.debug("{} Deleting unwanted directory {}", logPrefix, localPath.toAbsolutePath());
+                vertx.fileSystem().deleteRecursiveBlocking(localPath.toAbsolutePath().toString(), true);
+
+                directoriesDeleted.incrementAndGet();
+            }
+            else
+            {
+                LOGGER.debug("{} Deleting unwanted file: {}", logPrefix, localPath.toAbsolutePath());
+                vertx.fileSystem().deleteBlocking(localPath.toString());
+                filesDeleted.incrementAndGet();
+            }
+        }
+
+        LOGGER.info("{} Deleted {} unwanted files from local.", logPrefix, filesDeleted.get());
+        LOGGER.info("{} Deleted {} unwanted directories from local.", logPrefix, directoriesDeleted.get());
+    }
+
+    /**
+     * Shortlists list of files and directories to download from source.
+     *
+     * @param instanceFilesListResponse list of files response received from source.
+     * @param successThreshold          success threshold to determine whether downloader should continue to download or not.
+     * @return List of instance files to download. Returned list will be empty if successThreshold meets or no files remained to download.
+     */
+    @VisibleForTesting
+    Future<List<InstanceFileInfo>> shortlistDownloadFiles(InstanceFilesListResponse instanceFilesListResponse,
+                                                          double successThreshold)
+    {
+        List<InstanceFileInfo> filesToDownload =
+        instanceFilesListResponse.getFiles().stream()
+                                 .filter(instanceFileInfo -> !Files.exists(localPath(instanceFileInfo.fileUrl, instanceMetadata)))
+                                 .collect(Collectors.toList());
+
+        // filter out directories while calculating download size
+        long downloadSize = filesToDownload.stream()
+                                           .filter(instanceFileInfo -> instanceFileInfo.fileType.equals(FileType.FILE))
+                                           .map(instanceFileInfo -> instanceFileInfo.size)
+                                           .reduce(0L, Long::sum);
+
+        if (downloadSize == 0)
+        {
+            // downloadSize of zero means that there are no files to download OR
+            // only empty directories and/or empty files left to create.
+            return Future.succeededFuture(filesToDownload);
+        }
+
+        long totalSize = instanceFilesListResponse.getTotalSize();
+
+        // Calculate percentage of data required to download
+        double toDownloadPercentage = ((totalSize - downloadSize) * 1.0) / totalSize;
+        LOGGER.info("{} Remaining size to download={} bytes", logPrefix, downloadSize);
+        LOGGER.info("{} Download percentage={}, success threshold={}", logPrefix, toDownloadPercentage, successThreshold);
```

Comment:
1. combine both?
2. I suggest changing the field names in the log message. See below. The reason is to have the fieldName a single word (camel case to be consistent with other log message) in order to be extracted by log analysis tooling in a meaningful way. Please check the other log messages in the patch too. Also let's make sure to use the consistent filed name, so aggregation in log analysis makes sense. 

```suggestion
        LOGGER.info("{} Remaining size to download={} bytes. downloadedPercentage={} successThreshold={}", 
                    logPrefix, downloadSize, toDownloadPercentage, successThreshold);
```

Reply:
done